### ISSUE TITLE
Constructing n m-sized StellarGraphs is much slower than one (n*m)-sized one

### ISSUE THREAD

**Issue by huonw** [stellargraph/stellargraph#1336]
### Describe the bug

For graph classification, one is often constructing many small graphs, e.g. 10k graphs of 4-30 nodes. Constructing all of these as `StellarGraph`s seems to much slower than constructing a single StellarGraph of 40k-300k nodes.

(StellarGraph 1.0.0rc1 dogfooding)

**Comment by huonw** [comment]
This is likely going to be much improved by things like #1291.

**Comment by kieranricardo** [comment]
I tested this by comparing creating:

1.  10k graphs with 100 nodes and 300 edges
2. 1 graph with 1mil nodes and 3mil edges

Develop vs latest #1291:

1. 9s -> 0.4s
2. 64s -> 58s

So the absolute construction times goes down for both cases, but the relative slowdown is about 1000 times worse. The 58s is just from creating 10k empty stellargraphs.

**Comment by kieranricardo** [comment]
Ah looks like creating an empty pandas dataframe with three columns accounts for ~50% of the overhead of creating an empty stellargraph (2.6ms vs 5.7ms). I'm don't think there's much we can do about the overhead of creating a large number of python objects :(

**Comment by kieranricardo** [comment]
^ all timing done on #1291

**Comment by huonw** [comment]
One way would potentially be construction from NumPy more directly (and, with #1228, even TensorFlow tensors), and avoid some of the Pandas overhead.

**Comment by kieranricardo** [comment]
good idea! i thought numpy would have the same draw backs but creating an `np.empty(shape=(0,3))` is ~2000x faster than pandas
